osmapi.py

```diff
@@ -0,0 +1,126 @@
+import sys
+import xml.etree.cElementTree as ElementTree
+from oauthcli import OpenStreetMapAuth
+
+
+class HTTPError(Exception):
+    pass
+
+
+class OSM_API(object):
+    client_id = 'w2dn9_-3kWlROhfmsY8giTIC1XiewJCijZskbF8lltk'
+    client_secret = 'q2DThNZJcFBA12STvmvXOz9zMBdr0zvkCgiLkX31tp0'
```

Comment:
Shouldn't this secret be omitted from the source code? Usually such things have to be concealed in a separate configuration file provided by the user.

Reply:
In my experience it is actually quite common to find API keys in source code. It's not like we want every user to create their own key, it's often "one key per application" so it has to be somewhere in the application code -- which means in the open, for opensource code. The goal is just to track which application is making requests.